pkg/repository/udmrepo/kopialib/backend/file_system.go

```diff
@@ -0,0 +1,81 @@
+/*
+Copyright the Velero contributors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package backend
+
+import (
+	"context"
+	"os"
+	"path/filepath"
+	"strings"
+
+	"github.com/kopia/kopia/repo/blob"
+	"github.com/kopia/kopia/repo/blob/filesystem"
+	"github.com/pkg/errors"
+
+	"github.com/vmware-tanzu/velero/pkg/repository/udmrepo"
+)
+
+type FsBackend struct {
+	options filesystem.Options
+}
+
+const (
+	defaultFileMode = 0o600
+	defaultDirMode  = 0o700
+)
+
+func (c *FsBackend) Setup(flags map[string]string) error {
+	path, err := mustHaveString(udmrepo.StoreOptionFsPath, flags)
+	if err != nil {
+		return err
+	}
+
+	prefix := optionalHaveString(udmrepo.StoreOptionPrefix, flags)
+
+	c.options.Path = resolveUserFriendlyPath(filepath.Join(path, prefix), false)
+	c.options.FileMode = defaultFileMode
+	c.options.DirectoryMode = defaultDirMode
+
+	c.options.Limits = setupLimits(flags)
+
+	return nil
+}
+
+func (c *FsBackend) Connect(ctx context.Context, isCreate bool) (blob.Storage, error) {
+	if !filepath.IsAbs(c.options.Path) {
+		return nil, errors.Errorf("filesystem repository path must be absolute")
+	}
+
+	return filesystem.New(ctx, &c.options, isCreate)
+}
+
+func resolveUserFriendlyPath(path string, relativeToHome bool) string {
+	home, _ := os.UserHomeDir()
+	if home != "" && strings.HasPrefix(path, "~") {
```

Comment:
Why is this needed?  This may cause issue if the input `path` is like `~some/path` and it may also cause issues in cross OS support.
Can't we make it simpler to force it to be an absolute path?

Reply:
Done, removed the processing of relative path.